src/code/z_player_lib.c

```diff
@@ -2035,21 +2599,203 @@ void Player_DrawGetItem(PlayState* play, Player* player) {
                 refPos.y = player->actor.world.pos.y + 28.0f;
             }
         } else {
-            Math_Vec3f_Copy(&refPos, &D_801F59E8);
+            Math_Vec3f_Copy(&refPos, &sGetItemRefPos);
         }
 
-        drawIdPlusOne = ABS_ALT(player->getItemDrawId);
+        drawIdPlusOne = ABS_ALT(player->getItemDrawIdPlusOne);
         Player_DrawGetItemImpl(play, player, &refPos, drawIdPlusOne);
     }
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80126AB4.s")
+void func_80126AB4(Player* player, Vec3f** arg1) {
+    if ((player->transformation == PLAYER_FORM_GORON) || (player->actor.id == ACTOR_EN_TEST3)) {
+        Math_Vec3f_Copy(&player->unk_AF0[1], &player->meleeWeaponInfo[0].base);
+        *arg1 = D_801C09B8;
+        return;
+    }
+
+    D_801C0994[1].x = D_801C0994[0].x;
+
+    if (player->unk_ADD >= 3) {
+        player->unk_ADD++;
+        D_801C0994[1].x *= 1.0f + ((9 - player->unk_ADD) * 0.1f);
+    }
+
+    D_801C0994[1].x += 1200.0f;
+    D_801C0994[2].x = D_801C0994[1].x;
+    *arg1 = D_801C0994;
+}
+
+void func_80126B8C(PlayState* play, Player* player) {
+    Vec3f* sp1C;
+
+    func_80126AB4(player, &sp1C);
+    func_8012669C(play, player, sp1C, D_801C0970);
+}
+
+// Zora boomerangs (?)
+void func_80126BD0(PlayState* play, Player* player, s32 arg2) {
+    if ((player->transformation != PLAYER_FORM_ZORA) || (player->rightHandType == PLAYER_MODELTYPE_RH_HOOKSHOT)) {
+        return;
+    }
+
+    if ((arg2 != 0) && (player->stateFlags1 & PLAYER_STATE1_400000)) {
+        OPEN_DISPS(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80126B8C.s")
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_OPA_DISP++, object_link_zora_DL_0110A8);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80126BD0.s")
+        CLOSE_DISPS(play->state.gfxCtx);
+    } else {
+        Actor* boomerangActor = player->boomerangActor;
+        Vec3f sp58;
+        Vec3f sp4C;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801271B0.s")
+        if (player->stateFlags1 & PLAYER_STATE1_2000000) {
+            if (player->boomerangActor == NULL) {
+                return;
+            }
+            if ((player->boomerangActor->params == arg2) ||
+                (((boomerangActor->child != NULL)) && (boomerangActor->child->params == arg2))) {
+                return;
+            }
```

Comment:
I feel like this could be split in three parts
```c
if (player->boomerangActor->params == arg2) {
    return;
}
```
Or possibly merged into 1 early return:
```c
if ((player->boomerangActor == NULL) || ... )
```

Reply:
For some reason separating it into three parts doesn't match but merged into 1 big early return matches
https://decomp.me/scratch/Sy2or

I'm not a huge fan of having a lot of conditions in a single `if` but I could do it if you think it would look better